addons/bus/static/src/js/web_client_bus.js

```diff
@@ -0,0 +1,91 @@
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * Detects the presence of assets in DOM's HEAD
+         *
+         * @override
+         */
+        async start() {
+            this._assetsChangedNotificationId = null;
+            this._assets = {};
+            this.call('bus_service', 'onNotification', this, this._onNotification);
+            await this._super(...arguments);
+        },
+        /**
+         * Assigns handler to bus notification
+         *
+         * @override
+         */
+        show_application() {
+            const showed = this._super(...arguments);
+            document.querySelectorAll('*[data-asset-xmlid]').forEach(el => {
+                this._assets[el.getAttribute('data-asset-xmlid')] = el.getAttribute('data-asset-version');
+            });
+            this.call('bus_service', 'startPolling');
+            return showed;
+        },
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * Displays one notification on user's screen when assets have changed
+         *
+         * @private
+         */
+        _displayBundleChangedNotification() {
+            if (!this._assetsChangedNotificationId) {
+                this._assetsChangedNotificationId = this.call('notification', 'notify', {
+                    title: _t('Update available'),
+                    message: _t('Refresh your browser to take advantage of the latest update of Odoo.'),
+                    sticky: true,
+                    onClose: () => {
+                        this._assetsChangedNotificationId = null;
+                    },
+                    buttons: [{
+                        text: _t('Refresh'),
+                        primary: true,
+                        click: () => {
+                            window.location.reload(true);
+                        }
+                    }],
+                 });
+            }
+        },
+
+        //--------------------------------------------------------------------------
+        // Handlers
+        //--------------------------------------------------------------------------
+
+        /**
+         * Reacts to bus's notification
+         *
+         * @private
+         * @param {Array} notifications: list of received notifications
+         */
+        _onNotification(notifications) {
+            for (const notif of notifications) {
+                if (notif[0][1] === 'bundle_changed') {
+                    const bundleXmlId = notif[1][0];
+                    const bundleVersion = notif[1][1];
+                    if (bundleXmlId in this._assets && bundleVersion !== this._assets[bundleXmlId]) {
+                        this._displayBundleChangedNotification();
```

Comment:
I think it's important to add a random delay here.

Out of experience, when a bundle is updated it's often not alone (the code was updated and it may impact several bundles). And if the website is very busy (e.g. odoo.com) the server is suddenly hammered with requests trying to recompile assets in parallel. Some of them will even be rolled back, retried, etc. Basically a sudden storm of requests that can last for a variable time until things get stable again.

This means:
- users might get multiple "bundle_changed" notifications in a row, and if they refresh too soon they'll get the notification *again*
- if all users get the notification at the same time and refresh, they will amplify the hammering on the server, make the experience worse for everybody

If we delay the notification by some random delay between e.g. 10 and 60 seconds, we flatten the surge of requests and avoid double notifications ;-)